### PR TITLE
feat: add LiteFS example for Flipt in the examples directory

### PR DESCRIPTION
**Pull request by yquansah**
This PR adds a LiteFS example for running Flipt against it.

### CODE REVIEW

File: examples/database/litefs/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+## Running the Example
+
+1. Run `docker-compose up` from this directory
+1. Open the Flipt UI (default: [http://localhost:8080](http://localhost:8080))
+
+## Details
+
+`docker compose` will spin up two instances of Flipt with embedded SQLite databases. On top of these instances is an nginx proxy that will forward "write" requests, anything but `GET`, to the primary. `GET` requests will be served from the instance's embedded SQLite database.
+
+LiteFS talks about following this [caveat](https://fly.io/docs/litefs/proxy/#how-it-works) in order for it to work properly.
```
Comment (markphelps) [inline examples/database/litefs/README.md:21]:
```suggestion
LiteFS describes a few [caveats](https://fly.io/docs/litefs/proxy/#how-it-works) that must be kept in mind.
```

File: examples/database/litefs/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+## Requirements
+
+- [Docker](https://www.docker.com/)
+- [docker-compose](https://docs.docker.com/compose/install/)
+
+## Running the Example
+
+1. Run `docker-compose up` from this directory
+1. Open the Flipt UI (default: [http://localhost:8080](http://localhost:8080))
+
```
Comment (markphelps) [inline examples/database/litefs/README.md:16]: is there any way the user could see that the data is getting replicated? like we have in the libsql example? https://github.com/flipt-io/flipt/tree/main/examples/database/libsql

i realize we are routing the writes and reads to different instances, but its not evident when you load the ui. j/w

Comment (yquansah) [reply examples/database/litefs/README.md:16]: @markphelps Great question. I could expose both instances, but just didn't want users to accidentally try to write on the replica instance (you would get an error).

I could just make that clear in the docs though, will do that.

File: examples/database/litefs/nginx/nginx.conf [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    }
+    server {
+        listen 80;
+        location / {
+            # Basically all writes should go to this primary,
+            # while reads go to everything else.
+            if ($request_method ~ "(PUT|POST|PATCH|DELETE)") {
+                proxy_pass http://primary;
+            }
+            proxy_pass http://all;
```
Comment (GeorgeMac) [inline examples/database/litefs/nginx/nginx.conf:17]: Nginx is weird. I went down a rabbithole learning about `if`. Apparently it is imperative, but they tried to make it support some declarative cases and now they have a [monster](https://www.nginx.com/resources/wiki/start/topics/depth/ifisevil/).

I think what you have works, but it reads like it should proxy pass to `all` in all cases (if it was just evaluated imperatively top to bottom). Just thought I would share that observation.

**Review by markphelps — APPROVED** [review]

**Pull request merged**